### PR TITLE
[AAE-2200] Upload dialog title should match selected site.

### PR DESCRIPTION
**Pull request by BaptisteMahe**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
The update of the title doesn't match the selected site but the selected node and not every time depending on the context of the upload. Also when the site-dropdown receives a site guid as init value and the site isn't int the first loaded site list it doesn't select that site.

**What is the new behaviour?**
The title of an upload dialog is now matching its selected site inside the dropdown-site. Also at the initialisation is a start folder is selected the site-dropdown select the parent site of that folder (My files by default) which leads to the update of the title of the dialog.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://issues.alfresco.com/jira/browse/AAE-2200
*Open dialog with starting folder inside "My site 1"*
open-dialog-with-starting-folder-inside-my-site-1
*Open dialog without starting folder*
open-dialog-without-starting-folder
*Updated dialog after changing site using site-dropdown*
updated-dialog-after-changing-site

### CODE REVIEW

**Comment by DenysVuika** [comment]
"CHOOSE ... TO" is grammatically incorrect in this case. It can be "Move ... to" or "Copy ... to", but not "choose"

**Comment by BaptisteMahe** [comment]
> "CHOOSE ... TO" is grammatically incorrect in this case. It can be "Move ... to" or "Copy ... to", but not "choose"
Just had a chat with Mark, he is on it.

File: lib/content-services/src/lib/content-node-selector/content-node-dialog.service.ts [hunk 1]
```diff
@@ -179,7 +179,7 @@ export class ContentNodeDialogService {
         });
 
         const data: ContentNodeSelectorComponentData = {
-            title: this.getTitleTranslation(action, contentEntry.name),
+            title: this.getTitleTranslation(action, this.translation.instant('DROPDOWN.MY_FILES_OPTION')),
```
Comment (adinapitul) [inline lib/content-services/src/lib/content-node-selector/content-node-dialog.service.ts:182]: Are you sure this doesn't break the functionality of the dropdown in ADW on Copy / Move action?

File: lib/content-services/src/lib/content-node-selector/content-node-selector.component.ts [hunk 2]
```diff
@@ -47,21 +47,24 @@ export class ContentNodeSelectorComponent {
 
     onSelect(nodeList: Node[]) {
         this.chosenNode = nodeList;
-        this.updateTitle(nodeList);
+    }
+
+    onSiteChange(siteTitle: string) {
```
Comment (adinapitul) [inline lib/content-services/src/lib/content-node-selector/content-node-selector.component.ts:52]: Would this apply to ADW as well in Copy / Move action? If yes, this is not ok. In ADW in Copy / Move dialog, the dialog title should show the name of the item we're copying, and this title should NOT change.

File: lib/content-services/src/lib/content-node-selector/content-node-selector.component.ts [hunk 3]
```diff
@@ -47,21 +47,24 @@ export class ContentNodeSelectorComponent {
...
         this.data.select.complete();
     }
 
-    updateTitle(nodeList: Node[]): void {
-        if (this.action === 'CHOOSE' && nodeList) {
-            this.title = this.getTitleTranslation(this.action, nodeList[0].name);
+    updateTitle(siteTitle: string) {
+        if (this.action === 'CHOOSE' && siteTitle) {
+            this.title = this.getTitleTranslation(this.action, siteTitle);
         }
```
Comment (BaptisteMahe) [inline lib/content-services/src/lib/content-node-selector/content-node-selector.component.ts:64]: @adinapitul for your concerns about breaking the title in the Copy / Move dialog. We are checking here that the action is "CHOOSE" (action for upload dialog) and not "COPY" or "MOVE" before updating the title.

File: lib/content-services/src/lib/content-node-selector/content-node-selector-panel.component.ts [hunk 4]
```diff
@@ -248,6 +262,19 @@ export class ContentNodeSelectorPanelComponent implements OnInit, OnDestroy {
         this.onDestroy$.complete();
     }
 
+    private getStartSite() {
+        this.nodesApiService.getNode(this.currentFolderId).subscribe((startNodeEntry) => {
+            this.startSiteGuid = this.activitiContentService.getSiteNameFromNodePath(startNodeEntry);
```
Comment (eromano) [inline lib/content-services/src/lib/content-node-selector/content-node-selector-panel.component.ts:267]: I am not sure about the use of this service let's talk about it tomorrow

**Comment by adinapitul** [comment]
Why do we need the name of the folder in the dialog title?

Can we just have the title say something like "Select a file to attach..." / "Select one or more files to attach..." ?

**Pull request merged**